packages/app-root/src/hooks/usePanoptesUser.js

```diff
@@ -0,0 +1,56 @@
+import auth from 'panoptes-client/lib/auth'
+import { useEffect, useState } from 'react'
+
+import { fetchPanoptesUser } from '../helpers'
+
+const isBrowser = typeof window !== 'undefined'
+
+if (isBrowser) {
+  auth.checkCurrent()
+}
+
+const localStorage = isBrowser ? window.localStorage : null
+const storedUserJSON = localStorage?.getItem('panoptes-user')
+let user = storedUserJSON && JSON.parse(storedUserJSON)
+/*
+  Null users crash the ZooHeader component.
+  Set them to undefined for now.
+*/
```

Comment:
I’m not sure if it’s an issue.

Reply:
I think the default here should be null, not an empty object. Probably depends on what we expect `user` to be when no one is logged in. 

https://github.com/zooniverse/front-end-monorepo/blob/d2720c116dc2c51178587d29f6df78c88dc5d61b/packages/lib-react-components/src/ZooHeader/ZooHeader.js#L46

Reply:
With Server Components, it might even be better to pass a promise that resolves to a user. 

https://react.dev/reference/react/use#streaming-data-from-server-to-client

Reply:
Ah okay, I understand the need to set to `undefined` now - thanks!